drivers/tcs37727/tcs37727.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (C) 2015 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_tcs37727
+ * @{
+ *
+ * @file
+ * @brief       Driver for the AMS TCS37727 Color Light-To-Digital Converter
+ *
+ * @author      Felix Siebel <f.siebel@phytec.de>
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "periph/i2c.h"
+#include "tcs37727.h"
+#include "tcs37727-internal.h"
+
+#define ENABLE_DEBUG   (1)
+#include "debug.h"
+
+#define I2C_SPEED      I2C_SPEED_FAST
+
+static int tcs37727_test(tcs37727_t *dev)
+{
+    char id;
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_read_reg(dev->i2c, dev->addr, TCS37727_ID, &id) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (id != TCS37727_ID_VALUE) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int tcs37727_init(tcs37727_t *dev, i2c_t i2c, uint8_t address, int atime_us)
+{
+    /* write device descriptor */
+    dev->i2c = i2c;
+    dev->addr = address;
+    dev->initialized = false;
+
+    i2c_acquire(dev->i2c);
+
+    /* initialize the I2C bus */
+    if (i2c_init_master(i2c, I2C_SPEED) < 0) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (tcs37727_test(dev)) {
+        return -2;
+    }
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_write_reg(dev->i2c, dev->addr, TCS37727_CONTROL,
+                      TCS37727_CONTROL_AGAIN_4) != 1) {
+        i2c_release(dev->i2c);
+        return -3;
+    }
+    dev->again = 4;
```

Comment:
Typically it does not make a sense to make it adjustable here, because you can not know in what kind of environment the sersor is, when the init function is executed.
And ... dev->again will be adjusted automatically.